### ISSUE TITLE
Hyperliquid - Error setting leverage: Unable to set leverage

### ISSUE THREAD

**Issue by nikspz** [hummingbot/hummingbot#7024]
### Describe the bug

Error setting leverage 1 for MNT-USD: Unable to set leverage.

Reproduced for higher (5-20) leverage as well on v1 and v2 strategies

[logs_conf_perpetual_market_making_1.log](https://github.com/hummingbot/hummingbot/files/15290231/logs_conf_perpetual_market_making_1.log)
[logs_conf_perpetual_market_making_2.log](https://github.com/hummingbot/hummingbot/files/15290235/logs_conf_perpetual_market_making_2.log)
```
2024-05-13 05:33:18,380 - 4153484 - hummingbot.client.hummingbot_application - INFO - start command initiated.
2024-05-13 05:33:18,785 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2024-05-13 05:33:19,036 - 4153484 - hummingbot.core.data_type.order_book_tracker - INFO - Initialized order book for MNT-USD. 1/1 completed.
2024-05-13 05:33:19,042 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_api_order_book_data_source.HyperliquidPerpetualAPIOrderBookDataSource - INFO - Subscribed to public order book, trade channels...
2024-05-13 05:33:19,282 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_user_stream_data_source.HyperliquidPerpetualUserStreamDataSource - INFO - Subscribed to private order and trades changes channels...
2024-05-13 05:33:19,400 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - NETWORK - Error setting leverage 1 for MNT-USD: Unable to set leverage
2024-05-13 05:33:23,858 - 4153484 - hummingbot.core.rate_oracle.rate_oracle - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2024-05-13 05:33:28,011 - 4153484 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - Changing position mode to ONEWAY succeeded.
2024-05-13 05:33:28,087 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - NETWORK - Error setting leverage 1 for MNT-USD: Unable to set leverage
2024-05-13 05:33:29,002 - 4153484 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (MNT-USD) Creating 1 LIMIT bid orders at (Size, Price): ['15 MNT, 0.9655299999999999993605115378 USD'] to OPEN position.
2024-05-13 05:33:29,010 - 4153484 - hummingbot.strategy.perpetual_market_making.perpetual_market_making - INFO - (MNT-USD) Creating 1  LIMIT ask orders at (Size, Price): ['15 MNT, 0.9694000000000000394351218347 USD'] to OPEN position.
2024-05-13 05:33:29,193 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - NETWORK - Error submitting sell LIMIT order to Hyperliquid_perpetual for 15.0 MNT-USD 0.9694000000000000394351218346855603158473968505859375.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/derivative/hyperliquid_perpetual/hyperliquid_perpetual_derivative.py", line 400, in _place_order
    o_order_result = order_result['response']["data"]["statuses"][0]
TypeError: string indices must be integers
2024-05-13 05:33:29,203 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - WARNING - Failed to submit SELL order to Hyperliquid_perpetual. Check API key and network connection.
2024-05-13 05:33:29,211 - 4153484 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1715578409.0, "order_id": "0x116c1dc1021284de6ee4dea30c9896c7", "order_type": "OrderType.LIMIT", "event_name": "MarketOrderFailureEvent", "event_source": "hyperliquid_perpetual"}
2024-05-13 05:33:29,211 - 4153484 - hummingbot.connector.client_order_tracker - INFO - Order 0x116c1dc1021284de6ee4dea30c9896c7 has failed. Order Update: OrderUpdate(trading_pair='MNT-USD', update_timestamp=1715578409.0, new_state=<OrderState.FAILED: 6>, client_order_id='0x116c1dc1021284de6ee4dea30c9896c7', exchange_order_id=None, misc_updates=None)
2024-05-13 05:33:29,212 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - NETWORK - Error submitting buy LIMIT order to Hyperliquid_perpetual for 15.0 MNT-USD 0.9655299999999999993605115378159098327159881591796875.
Traceback (most recent call last):
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 452, in _create_order
    await self._place_order_and_process_update(order=order, **kwargs,)
  File "/home/nikita/development/hummingbot/connector/exchange_py_base.py", line 469, in _place_order_and_process_update
    exchange_order_id, update_timestamp = await self._place_order(
  File "/home/nikita/development/hummingbot/connector/derivative/hyperliquid_perpetual/hyperliquid_perpetual_derivative.py", line 400, in _place_order
    o_order_result = order_result['response']["data"]["statuses"][0]
TypeError: string indices must be integers
2024-05-13 05:33:29,213 - 4153484 - hummingbot.connector.derivative.hyperliquid_perpetual.hyperliquid_perpetual_derivative.HyperliquidPerpetualDerivative - WARNING - Failed to submit BUY order to Hyperliquid_perpetual. Check API key and network connection.
2024-05-13 05:33:29,218 - 4153484 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1715578409.0, "order_id": "0xfec9c8e1ce722ec208ab948434d7d3b8", "order_type": "OrderType.LIMIT", "event_name": "MarketOrderFailureEvent", "event_source": "hyperliquid_perpetual"}
2024-05-13 05:33:29,218 - 4153484 - hummingbot.connector.client_order_tracker - INFO - Order 0xfec9c8e1ce722ec208ab948434d7d3b8 has failed. Order Update: OrderUpdate(trading_pair='MNT-USD', update_timestamp=1715578409.0, new_state=<OrderState.FAILED: 6>, client_order_id='0xfec9c8e1ce722ec208ab948434d7d3b8', exchange_order_id=None, misc_updates=None)
```

### Steps to reproduce

1. connect hyperliquid_perpetual
2. create strategy using hyperliquid_perpetual
3. start strategy and review logs

### Release version

dev-1.28.0

### Type of installation

Source

**Comment by yancong001** [comment]
Looking at the hyperliquid sdk, I found that they have some updates to their api, I will fix this bug

**Comment by yancong001** [comment]
here https://github.com/hummingbot/hummingbot/pull/7027

**Comment by nikspz** [comment]
fixed, fix merged to latest development (dev-1.28.0)
